### ISSUE TITLE
[Top App Bar] Accessibility Focus Order Validation

### ISSUE THREAD

**Issue by ekager** [material-components/material-components-android#4693]
**Description:**
Our accessibility team filed bugs against our collapsing top app bars with the feedback that when using TalkBack the Navigation Icon should be ready before the title, but instead the title is read first. I tested with the vanilla Material components and found the same behavior here. Is this intentional? Do we have documentation around this decision? 

**Expected behavior:** Screenshots and/or description of expected behavior
Navigation Icon read first, then Title, then actions (LTR focusing?)

**Source code:** The code snippet which is causing this issue

**Minimal sample app repro:** Please consider attaching a minimal sample app that reproduces the issue. This will help narrow down the conditions required for reproducing the issue, and it will speed up the bug fix process. You may attach a zip file of the sample app or link to a GitHub repo that contains the sample app.
Material Components Catalog App -> Top App Bar -> Collapsing Demo (Medium)

**Material Library version:** Material Android Library version you are using here (e.g., 1.1.0-alpha07)
Main branch

**Device:** Device on which the bug was encountered here
Pixel 8 Pro API 34

**Comment by victorcatao** [comment]
+1
